advsecurenet/computer_vision/object_detection/attacks/adversarial_patch_based/dpatch.py

```diff
@@ -0,0 +1,592 @@
+"""
+This code was written based on the following sources:
+
+https://arxiv.org/pdf/1806.02299
+https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/art/attacks/evasion/dpatch.py
+"""
+
+import math
+import random
+import warnings
+import numpy as np
+import torch
+from tqdm.auto import trange, tqdm
+
+from advsecurenet.shared.types.configs.attack_configs.dpatch_attack_config import (
+    DPatchAttackConfig,
+)
+from advsecurenet.computer_vision.object_detection.attacks.base.object_detection_attack import (
+    ObjectDetectionAttack,
+)
+from advsecurenet.utils.device_utils import move_batch_to_device
+
+
+class DPatch(ObjectDetectionAttack):
+    """
+    DPatch attack
+
+    Paper: https://arxiv.org/abs/1712.04248
```

Comment:
Why is this listed here but not at the beginning of the file? Also, what is the difference from tps://arxiv.org/pdf/1806.02299 ?